### PR TITLE
CBMC Proof of initFileTransfer_Http function

### PR DESCRIPTION
**Pull request by Bujain**
*Description of changes:*

Added the files related to the CBMC proof of initFileTransfer_Http function 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    OtaHttpStatus_t status;
+    OtaInterfaces_t interface;
+    OtaAgentContext_t agent;
+
+    pAgentCtx = &agent;
+
+    /* Initialize the file context from the Agent context. */
+    pAgentCtx->fileContext = fileContext;
+
+    /* Updating the function pointer in Http to the stub. */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c:52]: You can remove these comments about setting a variable.

**Review by yanjos-dev — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c]: Updated it in 66e5b68

File: test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    OtaHttpInterface_t http;
+    OtaHttpStatus_t status;
+    OtaInterfaces_t interface;
+    OtaAgentContext_t agent;
+
+    pAgentCtx = &agent;
+
+    pAgentCtx->fileContext = fileContext;
+
+    /* The init reference in the http interface will always point to a function address and thus
```
Comment (yanjos-dev) [inline test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c:51]: How do we know this to be the case?

Comment (Bujain) [reply test/cbmc/proofs/initFileTransfer_Http/initFileTransfer_Http_harness.c:51]: I have added an assert statement to check if the function pointer is initialized or not.

**Review by yanjos-dev — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**